### PR TITLE
Document Google AMP Cache requirements.

### PR DESCRIPTION
**Pull request by twifkak**
Document the set of requirements that the Google AMP Cache sets on
signed exchanges, to enable independent signing implementations.

### CODE REVIEW

File: docs/cache_requirements.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+ * The signed `fallback URL` must equal the URL at which the SXG was delivered.
+ * The payload must be:
+   * non-empty.
+   * valid transformed AMP. The canonical definition of transformed AMP is the
+     return value of `transform.Process()`. If given a valid AMP doc as input,
+     it should produce a valid transformed AMP doc. There may be other ways of
+     achieving this, but they are unsupported (i.e. may arbitrarily break in the
+     future).
+   * unchanged after calling `transform -config NONE`.
```
Comment (alin04) [inline docs/cache_requirements.md]: Should this link to the cmd?

File: docs/cache_requirements.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+The Google AMP cache sets some requirements in addition to the ones set by the
+[SXG spec](https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html).
+These include:
+
+ * The signed `fallback URL` must equal the URL at which the SXG was delivered.
+ * The payload must be:
+   * non-empty.
+   * valid transformed AMP. The canonical definition of transformed AMP is the
+     return value of `transform.Process()`. If given a valid AMP doc as input,
```
Comment (alin04) [inline docs/cache_requirements.md]: Should this link to the source? At least the file?

Comment (alin04) [reply docs/cache_requirements.md]: Is there documentation on what "valid AMP" is? Does it make sense to link to the validator?

Comment (twifkak) [reply docs/cache_requirements.md]: Done and done. There is a [specification](https://amp.dev/documentation/guides-and-tutorials/learn/spec/amphtml) that seems out of date, so I linked to the validator instead.

File: docs/cache_requirements.md (hunk 1 above)
Comment (twifkak) [reply docs/cache_requirements.md]: Linked to the docs.

**Pull request merged**